bayesian/utility/v2/component_analyses.py

```diff
@@ -168,6 +168,50 @@ def get_batch_ca_data(ecosystem: str, packages: List) -> dict:
     return response
 
 
+def get_batch_ca_vulnerability_data(ecosystem: str, packages: List) -> dict:
+    """Fetch package details for component analyses."""
+    logger.debug('Executing get_batch_ca_data')
+    started_at = time.time()
+
+    response = None
+    semver_packages = []
+    pseudo_version_packages = []
+
+    # Need to seperate semver and pseudo verion packages for golang
+    if (ecosystem == "golang"):
+        for p in packages:
+            if p['is_pseudo_version']:
+                pseudo_version_packages.append(p)
+            else:
+                semver_packages.append(p)
+    else:
+        semver_packages = packages
+
+    graph_data_fetcher = []
+    if len(semver_packages) > 0:
+        get_semver_data = functools.partial(GraphAnalyses.get_batch_ca_vulnerability_data,
+                                            ecosystem)
+        graph_data_fetcher = list(_fetcher_in_batches(get_semver_data, semver_packages))
+
+    if len(pseudo_version_packages) > 0:
+        get_pseudo_data = functools.partial(GraphAnalyses.get_batch_ca_data_for_pseudo_version,
+                                            ecosystem)
+        graph_data_fetcher += list(_fetcher_in_batches(get_pseudo_data, pseudo_version_packages))
+
+    response = {
+        "result": {
+            "data": []
+        }
+    }
+    result = EXECUTOR.map(lambda f: f(), graph_data_fetcher)
+    for r in result:
+        response["result"]["data"] += r["result"]["data"]
+
+    elapsed_time = time.time() - started_at
+    logger.info("concurrent batch exec took %s sec", elapsed_time)
```

Comment:
is this required?

Reply:
yes.IT is present in our other apis too.

Reply:
how does it help?

Reply:
It tells us how long it took to execute a specific batch.

Reply:
is it recoded for future references?

Reply:
As you can see it is only getting logged. So its not recorded. :)

Reply:
Please look at the functionality of the api rather than this.These things can be removed or kept its not a major thing.

Reply:
if not recorded then remove it.

Reply:
ACK.

Reply:
i think we should keep it.It does not have any overhead and we get to see the execution time.In case of any discrepancy the log will we useful. Also we dont record execution duration of a post call.It will not serve any purpose.